### PR TITLE
dev-games/aseprite: add 1.3.5

### PR DESCRIPTION
**Pull request by winterheart**
Add new 1.3.5

### CODE REVIEW

File: dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch [hunk 1]
```diff
@@ -0,0 +1,149 @@
+--- a/laf/base/cfile.cpp
```
Comment (thesamesam) [inline dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch:1]: Is this upstream somewhre? If so, include a reference at the top? If not, why not?

Comment (thesamesam) [reply dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch:1]: ping

Comment (winterheart) [reply dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch:1]: Proposed upstream fix https://github.com/aseprite/laf/pull/84

Comment (ArsenArsen) [reply dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch:1]: please link to it above the patch, and git format-patch the patch properly (if git).  see, for instance, https://gitweb.gentoo.org/repo/gentoo.git/tree/app-emulation/qemu/files/qemu-7.2.0-linux-headers-6.2-glibc-2.36.patch

Comment (winterheart) [reply dev-games/aseprite/files/aseprite-1.3.5_laf-strict-aliasing.patch:7]: Done.

**Review by juippis — APPROVED** [review]
Sorry for the delay and thanks!

**Comment by eli-schwartz** [comment]
@juippis,

This PR reverts QA changes and shouldn't have been merged.

File: dev-games/aseprite/aseprite-1.3.5.ebuild [hunk 2]
```diff
@@ -96,19 +97,6 @@ src_prepare() {
...
-	# -Werror=strict-aliasing, -Werror=odr, -Werror=lto-type-mismatch
-	# https://bugs.gentoo.org/924692
-	# https://github.com/aseprite/aseprite/issues/4413
-	#
-	# There is a bundled skia that fails with ODR errors. When excluding just
-	# skia from testing, aseprite itself fails with strict-aliasing, and when
-	# that is disabled, fails again with ODR and lto-type-mismatch issues.
-	#
-	# There are a lot of issues, so don't trust any fixes without thorough
-	# testing.
```
Comment (eli-schwartz) [inline dev-games/aseprite/aseprite-1.3.5.ebuild:108]: Was this tested? I think not. At minimum, you haven't patched the QA violation in the skia archive, nor have you upgraded the skia revision.

**Comment by juippis** [comment]
This PR didn't revert anything, it added a new version with a fix **hopefully**. If the proposed fix doesn't work, I guess the old workaround can be re-applied.

**Comment by eli-schwartz** [comment]
The proposed fix fails to work, because the proposed fix solves strict-aliasing issues but not ODR issues.

Updating to a new version and removing a QA fix certainly appears to be reverting a fix. But @winterheart says I'm not allowed to add QA fixes to ebuilds after several months of an open ticket, because it is a "violation of maintainer autonomy", which looks like it has something to do with the removal of the QA fix.

I cannot possibly disagree more.

**Comment by juippis** [comment]
Okay, I see there's some history here. I don't look at git log category/package before checking these bumps. The `filter-lto` could've potentially been there for years (via filter-flags, and no, didn't look at diff for deleted file with a bug link). What parameters you want me to test the 1.3.5 with? `-Werror=strict-aliasing -Werror=odr -Werror=lto-type-mismatch`? I'll add the flags back if it fails.

My POV: 
 - Gentoo maintainer is aware of the issue (upstream PR link by the Gentoo maintainer),
 - upstream has an issue filed,
 - upstream has a PR to fix the problem.

@winterheart is a long-standing contributor with good reputation, so obviously I'm willing to trust they've tested the patch. We often end up carrying these "workarounds" (like redundant seds) in ebuilds, so it's nice to see an attempt cleaning them out.

And to @winterheart : I admit it would've been nice to see a PR with you tagged to review it, but the truth also is that some things just need to be pushed or otherwise they'd end up rotting/waiting for different maintainers. In my opinion the `filter-lto` is a rather simple commit, and the bug was open long enough to react. So I really don't see that much harm in it.

But since upstream is aware of the issue, that's THE best place to fix it ultimately. Let's work with each other, rather than against each other here?

**Comment by thesamesam** [comment]
Yes, certain work falls under broad-scope QA. In this case, it's about damage limitation (reducing risk of miscompilation). If maintainers want to do _further_ work to fix things properly, they can, although it might be a bit annoying sometimes if it blocks the LTO tracker when it's not strictly a blocker, even though it's related.

It's a shame that the additional Skia issue was missed, which Eli noted both on the bug and I would've expected @winterheart to spot when testing themselves, but mistakes happen.

**Comment by juippis** [comment]
So looks like with the patch `append-flags -fno-strict-aliasing` can be dropped, but `filter-lto` must be put back so `-flto -Werror=strict-aliasing -Werror=odr -Werror=lto-type-mismatch` will compile.

@eli-schwartz do you want me to author you, or should I just push it? And do you agree with this?

**Comment by eli-schwartz** [comment]
Yes, the patch should allow dropping -fno-strict-aliasing. I haven't tested it myself but I'm more than happy to take that on faith.

I know that `-flto -Werror=strict-aliasing -Werror=odr -Werror=lto-type-mismatch` will fail due to odr and lto-type-mismatch since I only added it a couple days ago and because I tested it with the upstream 1.3.5 and 1.3.6 releases (but didn't submit the updated versions, *specifically* to avoid stepping on the toes of the active maintainer -- I'm acting as an agent of QA here, and that begins and ends with LTO fixes).

Even 1.3.6 has the LTO errors upstream, and the patch doesn't include LTO fixes, only strict-aliasing fixes.

> @eli-schwartz do you want me to author you, or should I just push it? And do you agree with this?

I don't mind one way or another who the author is. If the comment (after tweaking to adapt to strict-aliasing being obsolete) is substantially the same as mine, I suppose a reasonable argument can be made for Co-authored-by?

And yes -- re-adding the filter-lto on its own sounds correct.

**Comment by thesamesam** [comment]
I came across https://github.com/aseprite/laf/pull/84#issuecomment-2100417928 -- could you check it? Do we need to drop the patch?

**Comment by eli-schwartz** [comment]
There was some discussion about it in #38133 but unfortunately the maintainer appears to have lost interest in the package once being met with review comments? See also https://bugs.gentoo.org/924692#c6

**Pull request closed without merging**